### ISSUE TITLE
Typo in the doc for `BatchMemoryManager`

### ISSUE THREAD

**Issue by lxuechen** [pytorch/opacus#321]
In the docstring of `BatchMemoryManager`, optimizer is misspelled as optimizeer (see below). 

"""
...
    If it was not the last, ``optimizer.step()`` will only clip per sample gradients and
    sum them into ``p.summed_grad`.` ``optimizeer.zero_grad()`` will clear ``p.grad_sample``,
    but will leave ``p.grad`` and ``p.summed_grad``
...
"""

**Comment by alexandresablayrolles** [comment]
Thanks for flagging, closing the issue as the fix is on the way.
